server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/package.scala

```diff
@@ -10,4 +12,23 @@ package object internal {
     def toHeaderSeq: List[Header] =
       underlying.asScala.map(e => Header(e.getKey, e.getValue)).toList
   }
+
+  implicit class RichChannelFuture(val cf: ChannelFuture) {
```

Comment:
Minor: why not declare it as value class?

Reply:
I agree it's a good idea to make it a value class, but what's the value added in making the `val` private?

Reply:
Why would you want to make the wrapped value to become an extension member? Do you ever want to call `someChannelFuture.cf`?

Reply:
There's a cost of increased verbosity in such a defensive approach, which isn't always worthwhile. If the value added is just "preventing possibility of calling `someChannelFuture.cf`" it doesn't seem to be any risk and I would lean towards keeping lower verbosity.

Reply:
Whatever, that's minor, but I always thought that

```scala
implicit class Extensions(private val thing: Thing) extends AnyVal {
  // extensions
}
```

is one of the most standard idioms in Scala. You can see it all over the standard library etc.
Adding a single keyword doesn't look like verbosity to me.

Reply:
I agree that the standalone `val` has no point (it's copied code), so I'm happy to make the class an `AnyVal`. As for verbosity, I think it's beneficial to have a bit less code, without losing/risking anything. However, a standard practice of making val  private here is also OK for me and it looks like Tapir has a few such cases already, so I'll follow the convention here as well. I'll also apply this to `NettyServerHandler` where it was copied from.

Reply:
> without losing/risking anything
If this was a part of public library API with backwards compatibility contract, it would be much worse - the accidentally exposed field could never be removed or renamed without a proper deprecation cycle. That's why I think it's better to be as conservative as possible with what is exposed as public.

Reply:
Good point with the binary compatibility. These particular internals aren't meant become public API, but sometimes stuff gets copy-pasted (as we're seeing here :) and it may indeed be better to stay a bit more defensive just in case.